src/gptscript.ts

```diff
@@ -320,6 +320,70 @@ export class GPTScript {
         return this._load({toolDefs, disableCache, subTool})
     }
 
+    async listCredentials(context: Array<string>, allContexts: boolean): Promise<Array<Credential>> {
+        if (!this.ready) {
+            this.ready = await this.testGPTScriptURL(20)
+        }
+        const resp = await fetch(`${GPTScript.serverURL}/credentials`, {
+            method: "POST",
+            body: JSON.stringify({context, allContexts})
+        })
```

Comment:
Does `runSubcommand` not work for these?

Reply:
I remember trying it and it not working for some reason, but I don't remember why...let me try that again and see.

Reply:
Taking a look at the code again, `RunSubcommand` doesn't let me specify an arbitrary request body to use. I can only pass it a tool and some run options. But I need to be able to pass things like a credential (in the case of create), or name/context information for the other operations.

Reply:
Let me know if you want me to change this or leave it as is @thedadams

Reply:
Can you try something like thhis?
```
const r: Run = new RunSubcommand("credentials", undefined, {}, GPTScript.serverURL)

r.request(payload)
await r.text()
```

Reply:
That worked, thanks. This is updated now